tools/src/test/java/org/neo4j/tools/txlog/CheckTxLogsTest.java

```diff
@@ -695,24 +754,43 @@ private static PropertyRecord propertyRecord( long id, boolean inUse, long prevP
 
     private static class CapturingInconsistenciesHandler implements InconsistenciesHandler
     {
-        List<Inconsistency> inconsistencies = new ArrayList<>();
+        List<RecordInconsistency> recordInconsistencies = new ArrayList<>();
+        List<TransactionInconsistencyCount> transactionInconsistencyCounts = new ArrayList<>();
+
+        @Override
+        public void reportInconsistentRecord( LogRecordAndVersion<?> committed, LogRecordAndVersion<?> current )
+        {
+            recordInconsistencies.add( new RecordInconsistency( committed, current ) );
+        }
 
         @Override
-        public void handle( LogRecord<?> committed, LogRecord<?> current )
+        public void reportInconsistentTransaction( long txId, long countOfInconsistencies )
         {
-            inconsistencies.add( new Inconsistency( committed, current ) );
+            transactionInconsistencyCounts.add( new TransactionInconsistencyCount( txId, countOfInconsistencies ) );
         }
     }
 
-    private static class Inconsistency
+    private static class RecordInconsistency
     {
-        final LogRecord<?> committed;
-        final LogRecord<?> current;
+        final LogRecordAndVersion<?> committed;
+        final LogRecordAndVersion<?> current;
 
-        Inconsistency( LogRecord<?> committed, LogRecord<?> current )
+        RecordInconsistency( LogRecordAndVersion<?> committed, LogRecordAndVersion<?> current )
         {
             this.committed = committed;
             this.current = current;
         }
     }
+
+    private static class TransactionInconsistencyCount
+    {
+        final long txId;
+        final long count;
+
+        TransactionInconsistencyCount( long txId, long count )
+        {
+            this.txId = txId;
+            this.count = count;
+        }
+    }
```

Comment:
Yes that would be nice, but harder to implement, and this should be sufficient for investigative purposes. It gives you the starting point for where you should start looking, usually backwards.